python/lsst/daf/butler/direct_query_driver/_driver.py

```diff
@@ -715,11 +717,11 @@ def apply_query_projection(
                         unique_keys.append(builder.joiner.fields[dataset_type]["collection_key"])
                     else:
                         derived_fields.append((dataset_type, "collection_key"))
-                elif dataset_field == "timespan" and plan.datasets[dataset_type].is_calibration_search:
+                elif dataset_field == "timespan" and is_calibration_search:
```

Comment:
This `dataset_field == "timespan"` case was already not covered before I got here.  I wanted to add a test but can't figure out what circumstances would trigger it, since the validity range comparison is handled in-DB, and DatasetRefs don't include the calibration collection timespan field.

Reply:
Right, this can't be tested until we add support for "general" results.